src/axom/sina/core/Record.cpp

```diff
@@ -101,7 +101,10 @@ void Record::addRecordAsLibraryData(Record const &childRecord, std::string const
       add(file);
     }
   }
-  auto newLibData = addLibraryData(name, childRecord.toNode());
+  // Note: the toNode is being used as an intermediary here. As such, we force the CurveOrder to be oldest
+  // first, aka "don't reorder". If the user wants the curves in a specific order, that's decided at dump time,
+  // else the Record will have no way to know the "original" order within the library.
+  auto newLibData = addLibraryData(name, childRecord.toNode(CurveSet::CurveOrder::REGISTRATION_OLDEST_FIRST));
```

Comment:
Reviewer context: library data represents a sort of "sub-record", specifically info a library might be responsible for that should be available with the host code's.